src/bus/policy.c

```diff
@@ -691,9 +697,11 @@ int policy_registry_import(PolicyRegistry *registry, CDVar *v) {
 
         c_dvar_read(v, "]bs)>", &apparmor, &bustype);
 
-        if (apparmor)
-                /* XXX: AppArmor is currently not supported. */
-                return POLICY_E_INVALID;
+        if (apparmor) {
+                r = bus_apparmor_set_bus_type(registry->apparmor, bustype);
+                if (r)
+                        return error_trace(r);
+        }
```

Comment:
We technically want the launcher and broker to be independent. Imagine they are in different projects, and the launcher has apparmor enabled, but the broker has not. In this case, the launcher would forward the information to the broker even in `enforced/required` mode, but the broker would then silently ignore it because the `bus_apparmor_set_bus_type()` fallback just returns success unconditionally.

Maybe this is a misconfiguration, and the launcher needs to ensure that the broker it runs with has apparmor enabled. But in this case, there is no point in passing the `apparmor-boolean` at all. We can just always call `bus_apparmor_set_bus_type()` and drop the boolean entirely. I think I would go with that initially, and rely on launchers to depend on correctly compiled/configured brokers.

Reply:
Note that SELinux does not suffer from the same problem, because it does not have the `required` mode as part of dbus-configuration. It simply relies on a correctly configured system. Though, it does have the `if_selinux_enabled` condition in configuration parsing, which again is really weird if the launcher and broker compile-time-configuration mismatch. Umpf.

Reply:
Ahh, no, I am wrong. If apparmor=disabled, we don't want to call set_bus_type(), I guess? This is weird. It means you need to check for `bustype == NULL` in your `*_check_*()` functions, and return success if no bustype is set (because it means apparmor was not configured. Right?

Reply:
After removing bustype parsing NULL is a valid configuration IIUIC, so I suppose always calling `bus_apparmor_set_bus_type()` makes sense and then I need to also store the `bool apparmor` in the registry, so that I can add `if (!apparmor_enabled) return success;` to the check functions.

Reply:
This sounds fine to me. I assume NULL would be a missing <bustype> declaration? I'd be fine using an empty string then, I don't see why we would need to distinguish it from a missing declaration?

Reply:
Yes, NULL would be a missing declaration. Rewriting that into an empty string should be ok and then NULL could indeed be used to check if AA is disabled.